pillarbox-player/src/main/java/ch/srgssr/pillarbox/player/analytics/metrics/MetricsCollector.kt

```diff
@@ -0,0 +1,312 @@
+/*
+ * Copyright (c) SRG SSR. All rights reserved.
+ * License information is available from the LICENSE file.
+ */
+package ch.srgssr.pillarbox.player.analytics.metrics
+
+import androidx.media3.common.C
+import androidx.media3.common.Format
+import androidx.media3.common.Player
+import androidx.media3.common.Timeline.Window
+import androidx.media3.exoplayer.DecoderCounters
+import androidx.media3.exoplayer.DecoderReuseEvaluation
+import androidx.media3.exoplayer.analytics.AnalyticsListener
+import androidx.media3.exoplayer.analytics.AnalyticsListener.EventTime
+import androidx.media3.exoplayer.source.LoadEventInfo
+import androidx.media3.exoplayer.source.MediaLoadData
+import ch.srgssr.pillarbox.player.PillarboxExoPlayer
+import ch.srgssr.pillarbox.player.analytics.PillarboxAnalyticsListener
+import ch.srgssr.pillarbox.player.analytics.PlaybackSessionManager
+import ch.srgssr.pillarbox.player.analytics.TotalPlaytimeCounter
+import ch.srgssr.pillarbox.player.analytics.extension.getUidOfPeriod
+import ch.srgssr.pillarbox.player.source.PillarboxMediaSource
+import ch.srgssr.pillarbox.player.utils.DebugLogger
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.milliseconds
+
+/**
+ * Playback stats metrics
+ * Compute playback stats metrics likes stalls, playtime, bitrate, etc..
+ */
+class MetricsCollector(
+    private val player: PillarboxExoPlayer,
+    private val timeProvider: () -> Long = { System.currentTimeMillis() }
+) : PillarboxAnalyticsListener, PlaybackSessionManager.Listener {
+
+    /**
+     * Listener
+     */
+    interface Listener {
+        /**
+         * On metric session finished
+         *
+         * @param metrics The [PlaybackMetrics] that belong to te finished session.
+         */
+        fun onMetricSessionFinished(metrics: PlaybackMetrics) = Unit
+
+        /**
+         * On metric session ready
+         *
+         * @param metrics
+         */
+        fun onMetricSessionReady(metrics: PlaybackMetrics) = Unit
```

Comment:
Shouldn't we use `metrics` here, to match the `PlaybackMetrics` name?
```suggestion
        /**
         * On metrics session finished
         *
         * @param metrics The [PlaybackMetrics] that belong to te finished session.
         */
        fun onMetricsSessionFinished(metrics: PlaybackMetrics) = Unit

        /**
         * On metrics session ready
         *
         * @param metrics
         */
        fun onMetricsSessionReady(metrics: PlaybackMetrics) = Unit
```